backend/api/Controllers/RobotController.cs

```diff
@@ -674,4 +770,34 @@ private async void OnIsarUnavailable(Robot robot)
         robot.CurrentMissionId = null;
         await _robotService.Update(robot);
     }
+
+    private static Pose ClosestSafePosition(Pose robotPose, IList<SafePosition> safePositions)
+    {
+        if (safePositions == null || safePositions.Count == 0)
+        {
+            throw new ArgumentException("Poses list cannot be null or empty.");
+        }
+
+        var closestPose = safePositions[0].Pose;
+        float minDistance = CalculateDistance(robotPose, closestPose);
+
+        for (int i = 1; i < safePositions.Count; i++)
+        {
+            float currentDistance = CalculateDistance(robotPose, safePositions[i].Pose);
+            if (currentDistance < minDistance)
+            {
+                minDistance = currentDistance;
+                closestPose = safePositions[i].Pose;
+            }
+        }
+        return closestPose;
+    }
+
+    private static float CalculateDistance(Pose pose1, Pose pose2)
+    {
+        var pos1 = pose1.Position;
+        var pos2 = pose2.Position;
+        return (float)Math.Sqrt(Math.Pow(pos1.X - pos2.X, 2) + Math.Pow(pos1.Y - pos2.Y, 2) + Math.Pow(pos1.Z - pos2.Z, 2));
+    }
```

Comment:
Maybe we should create an issue to calculate closest safe position based on cost map and A*

Reply:
Requires that we have the cost map available for all robots?

Reply:
Good point